packages/flutter/lib/src/widgets/nested_scroll_view.dart

```diff
@@ -863,3 +1141,503 @@ class _NestedOuterBallisticScrollActivity extends BallisticScrollActivity {
+  /// value during that layout.
+  double get layoutExtent => _layoutExtent;
+  double _layoutExtent;
+
+  /// The total scroll extent of the gap being absorbed by the
+  /// [SliverOverlapAbsorber].
+  ///
+  /// This corresponds to the [SliverGeometry.scrollExtent] of the child of the
+  /// [SliverOverlapAbsorber].
+  ///
+  /// This is updated during the layout of the [SliverOverlapAbsorber]. It
+  /// should not change at any other time. No notifications are sent when it
+  /// changes; clients (e.g. [SliverOverlapInjector]s) are responsible for
+  /// marking themselves dirty whenever this object sends notifications, which
+  /// happens any time the [SliverOverlapAbsorber] might subsequently change the
+  /// value during that layout.
+  double get scrollExtent => _scrollExtent;
+  double _scrollExtent;
+
+  void _setExtents(double layoutValue, double scrollValue) {
+    assert(_writers == 1, 'Multiple RenderSliverOverlapAbsorbers have been provided the same SliverOverlapAbsorberHandle.');
+    _layoutExtent = layoutValue;
+    _scrollExtent = scrollValue;
+  }
+
+  void _markNeedsLayout() => notifyListeners();
+
+  @override
+  String toString() {
+    String extra;
+    switch (_writers) {
+      case 0:
+        extra = ', orphan';
+        break;
+      case 1:
+        // normal case
+        break;
+      default:
+        extra = ', $_writers WRITERS ASSIGNED';
+        break;
+    }
+    return '$runtimeType($layoutExtent$extra)';
+  }
+}
+
+/// A sliver that wraps another, forcing its layout extent to be treated as
+/// overlap.
+///
+/// The difference between the overlap requested by the [child] sliver and the
+/// overlap reported by this widget, called the _absorbed overlap_, is reported
+/// to the [SliverOverlapAbsorberHandle], which is typically passed to a
+/// [SliverOverlapInjector].
+///
+/// See also:
+///
+///  * [NestedScrollView], whose documentation has sample code showing how to
+///    use this widget.
+class SliverOverlapAbsorber extends SingleChildRenderObjectWidget {
+  /// Creates a sliver that absorbs overlap and reports it to a
+  /// [SliverOverlapAbsorberHandle].
+  ///
+  /// The [handle] must not be null.
+  ///
+  /// The [child] must be a sliver.
+  const SliverOverlapAbsorber({
+    Key key,
+    @required this.handle,
+    Widget child,
+  }) : assert(handle != null),
+       super(key: key, child: child);
+
+  /// The object in which the absorbed overlap is recorded.
+  ///
+  /// A particular [SliverOverlapAbsorberHandle] can only be assigned to a
+  /// single [SliverOverlapAbsorber] at a time.
+  final SliverOverlapAbsorberHandle handle;
+
+  @override
+  RenderSliverOverlapAbsorber createRenderObject(BuildContext context) {
+    return new RenderSliverOverlapAbsorber(
+      handle: handle,
+    );
+  }
+
+  @override
+  void updateRenderObject(BuildContext context, RenderSliverOverlapAbsorber renderObject) {
+    renderObject
+      ..handle = handle;
```

Comment:
Cascade operator looks kind of weird here if you're only setting one thing.

Reply:
Yeah. This is done for consistence with all the other updateRenderObject methods in the framework.